added showInList (int default 1) database field.

BenderskiyMA opened the pull request:
added AllSensors resource for /api/sensors/all endpoint
changed /api/sensors behavior to get only showInList=1 records
added RUN python -m pip install --upgrade pip to Dockerfile
changed sensor model, added property showInList and constructor parameter. changed json() method
changed Sensor resource PUT and POST methods
tested in Postman

todo=>change base image.

BenderskiyMA committed:
added showInList (int default 1) database field.
added AllSensors resource for /api/sensors/all endpoint
changed /api/sensors behavior to get only showInList=1 records
added RUN python -m pip install --upgrade pip to Dockerfile
changed sensor model, added property showInList and constructor parameter. changed json() method
changed Sensor resource PUT and POST methods
tested in Postman

todo=>change base image.

gr13 commented on model/sensor.py:
```diff
@@ -16,6 +15,7 @@ class SensorModel(db.Model):
     lastGoodValue = db.Column(db.Float(precision=2), nullable=False, default=0.00)
     lastGoodValueTime = db.Column(db.TIMESTAMP, default=db.func.current_timestamp(), nullable=False)
     updateRate = db.Column(db.Integer, default=1)
+    showInList = db.Column(db.Integer, default=1)
```
Is it a Boolean property?

BenderskiyMA replied:
yes. it was the error with Boolean type yesterday))
chaged to Boolean.

gr13 replied:
fix the error)

gr13 commented on model/sensor.py:
```diff
@@ -27,7 +27,8 @@ def __init__(self, sensorName: str,
                  sensorIdentifier: str,
                  sensorType: str,
-                 updateRate: int):
+                 updateRate: int,
```
use data: Dict

gr13 replied:
of **kwargs https://realpython.com/python-kwargs-and-args/

BenderskiyMA replied:
use data: Dict
it is most safe to any changes...
but it will be source of many problem in the future))

gr13 replied:
add 
from typing import List
at the beginning of the file

BenderskiyMA replied:
> from typing import List
in this module i don't have List variables))

gr13 commented on model/sensor.py:
```diff
@@ -38,6 +39,7 @@ def __init__(self, sensorName: str,
         self.updateRate = updateRate
+        self.showInList = showInList
 
     def json(self):
```
use:
def json(self) -> Dict:

BenderskiyMA replied:
ok

gr13 commented on model/sensor.py:
```diff
@@ -89,6 +92,10 @@ def find_by_location_id(cls, _locationid: int):
     def find_all(cls):
```
Why don't you add docstrings?

BenderskiyMA replied:
added several

gr13 replied:
Doc string is to describe what the function is going to do, like

Returns JSON representation of the class
Returns all rows in the table

BenderskiyMA replied:
added docstrings into model\sensor.py

gr13 commented on resources/sensors.py:
```diff
@@ -7,6 +7,22 @@
+    def get(self) -> Tuple:
+        # if not is_admin():
+        #    return ERROR_ACCESS_DENIED, 403
+        sensors: list = [x.json() for x in SensorModel.find_all_except_hidden()]
```
sensors: List

BenderskiyMA replied:
ok

BenderskiyMA committed:
Some cosmetic changes.
Added several docstrings
Changed SensorModel __init__ parameters.
Changed showInList type (Integer->Boolean)

BenderskiyMA committed:
added docstrings.
added method find_by_location_id_except_hidden
